app/brave_main_delegate.cc

```diff
@@ -49,6 +49,20 @@
 #include "components/dom_distiller/core/dom_distiller_switches.h"
 #endif
 
+namespace {
+// staging "https://sync-v2.bravesoftware.com/v2" can be overriden by
+// switches::kSyncServiceURL manually
+#if defined(OFFICIAL_BUILD)
+// production
+const char kBraveSyncServiceURL[] = "https://sync-v2.brave.com/v2";
+#else
+// For local server development "http://localhost:8295/v2 can also be overriden
+// by switches::kSyncServiceURL
+// dev
+const char kBraveSyncServiceURL[] = "https://sync-v2.brave.software/v2";
```

Comment:
these two urls should be added to https://github.com/brave/brave-browser/blob/master/lib/whitelistedUrlPrefixes.js

Reply:
Will add these three
production "https://sync-v2.brave.com/v2"
staging "https://sync-v2.bravesoftware.com/v2"
dev "https://sync-v2.brave.software/v2"

staging will only be used when it is overridden by switch.

Reply:
issue: https://github.com/brave/brave-browser/issues/10192
pr: https://github.com/brave/brave-browser/pull/10180